src/components/Editor/EditorMenu.js

```diff
@@ -75,14 +78,19 @@ function getMenuItems(
     event
   );
 
-  const pairedType = isOriginalId(selectedLocation.sourceId)
-    ? L10N.getStr("generated")
-    : L10N.getStr("original");
+  const isOriginal = isOriginalId(selectedLocation.sourceId);
+  const isPrettyTabOpen =
+    !isOriginal && tabs.includes(getPrettySourceURL(selectedSource.get("url")));
+  const hasSourceMap = selectedSource.toJS().sourceMapURL;
+  const isPrettyPrinted = isPretty(selectedSource.toJS());
 
   const jumpLabel = {
     accesskey: L10N.getStr("editor.jumpToMappedLocation1.accesskey"),
-    disabled: false,
-    label: L10N.getFormatStr("editor.jumpToMappedLocation1", pairedType),
+    disabled: isOriginal || hasSourceMap ? false : !isPrettyTabOpen,
```

Comment:
Can we use `!isPrettyPrinted` here instead of `!isPrettyTabOpen`
then we won't need line 82??